src/npm-dependency-version-checker.js

```diff
@@ -3,19 +3,41 @@
 const resolve = require('resolve');
 const DependencyVersionChecker = require('./dependency-version-checker');
 
+const ALLOWED_ERROR_CODES = [
+  // resolve package error codes
+  'MODULE_NOT_FOUND',
+
+  // Yarn PnP Error Codes
+  'UNDECLARED_DEPENDENCY',
+  'MISSING_PEER_DEPENDENCY',
+  'MISSING_DEPENDENCY',
+];
+
+let pnp;
+
+try {
+  pnp = require('pnpapi');
+} catch (error) {
+  // not in Yarn PnP; not a problem
+}
+
 class NPMDependencyVersionChecker extends DependencyVersionChecker {
   constructor(parent, name) {
     super(parent, name);
 
     let addon = this._parent._addon;
 
+    let target = this.name + '/package.json';
+    let basedir = addon.root;
+
     let jsonPath;
+
     try {
-      jsonPath = resolve.sync(this.name + '/package.json', {
-        basedir: addon.root,
-      });
+      jsonPath = pnp
+        ? pnp.resolveToUnqualified(target, basedir)
+        : resolve.sync(this.name + '/package.json', { basedir });
```

Comment:
this can reuse target

Reply:
D'oh! Yes, you are right (can't believe I forgot to do that :( )